database/database.go

```diff
@@ -94,3 +94,24 @@ type Database interface {
 	io.Closer
 	health.Checker
 }
+
+// BlockDatabase defines the interface for storing and retrieving blockchain blocks.
+type BlockDatabase interface {
+	// WriteBlock inserts a block into the store at the given height with the specified header size.
+	WriteBlock(height uint64, block []byte, headerSize uint32) error
+
+	// ReadBlock retrieves a block by its height.
+	ReadBlock(height uint64) ([]byte, error)
+
+	// ReadHeader retrieves only the header portion of a block by its height.
```

Comment:
In avalanchego we also need to [just read the block ID](https://github.com/ava-labs/avalanchego/blob/c202c5aed1b04441012089c4fa411c4cdc19578a/snow/engine/snowman/block/vm.go#L59), which is used to service block dissemination.

I guess this header is only used for C-chain, but probably we should have yet another API call for the block ID...?